### PR TITLE
[MacOs] Slider inside StackLayout Horizontal Orientation crash

### LINKED ISSUE #3610: [MacOs]Slider inside StackLayout Horizontal Orientation
##Summary
When you put a label and a slider inside a StackLayout with Horizontal orientation in MacOs it breaks the page drawing when requesting for width. (NaN is not a valid value for width)

##API Changes

Fix the slider width setting when inside a horizontal orientated StackLayout.

##Intended Use Case
Use for a sliding option with description.

### PR DESCRIPTION
**Pull request by AndreiMisiukevich**
### Description of Change ###

Fix the slider width setting when inside a horizontal orientated StackLayout. (Default size)

### Issues Resolved ### 

- fixes #3610 

### API Changes ###
None

### Platforms Affected ### 
- MacOS

### Before/After Screenshots ### 
No crash

### Testing Procedure ###
Use sample from original issue ticket which provided by @hartez 

```xaml
<?xml version="1.0" encoding="utf-8"?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms" xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" xmlns:local="clr-namespace:_3610" x:Class="_3610.MainPage">
    <StackLayout Orientation="Horizontal">
        
        <Label Text="Welcome to Xamarin.Forms!" HorizontalOptions="Center" VerticalOptions="CenterAndExpand" />
        
        <Slider/>
        
    </StackLayout>
</ContentPage>
```

### CODE REVIEW

File: Xamarin.Forms.Platform.MacOS/Controls/FormsNSSlider.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	internal class FormsNSSlider : NSSlider
+	{
+		readonly CGSize _fitSize;
+
+		internal FormsNSSlider() : base(CGRect.Empty)
+		{
+			Continuous = true;
+			SizeToFit();
+			var size = Bounds.Size;
+			_fitSize = new CGSize(size.Width > 0 ? size.Width : 100, size.Height > 0 ? size.Height : 20);
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.MacOS/Controls/FormsNSSlider.cs]: Where did we got that 100x20 ? Is that on Apple documentation ? we should add some comments here about it.

Comment (AndreiMisiukevich) [reply Xamarin.Forms.Platform.MacOS/Controls/FormsNSSlider.cs]: @rmarinho frankly speaking, i pull this data from my brain 

```
var size =Bounds.Size; 
// this line returns (0, 19), so, i presume, that 19 is default height
```
but this topic says that default size of thumb is 21x21 
https://discussions.apple.com/thread/4004362
Maybe it makes sense to set default height as 21, but i haven't found any info about default width of these control

Comment (AndreiMisiukevich) [reply Xamarin.Forms.Platform.MacOS/Controls/FormsNSSlider.cs]: screenshot at feb 12 22-07-46

Seems, found :)

http://www.youtube.com/watch?v=JdwVTX9aCtU&t=0m20s

in this video (i attached timecode, 20th sec) that guy uses drag and drop for adding NsSlider, where we can see, that default width = 96 and height 21 indeed.

I will update PR

**Comment by AndreiMisiukevich** [comment]
@rmarinho updated :)

**Comment by rmarinho** [comment]
@AndreiMisiukevich can we just add a comment and maybe a link to some doc, it will help someone figure it out in the future, thanks.

**Comment by AndreiMisiukevich** [comment]
> @AndreiMisiukevich can we just add a comment and maybe a link to some doc, it will help someone figure it out in the future, thanks.

You mean comment here (Github) or in source code?
Thx

**Comment by rmarinho** [comment]
> > @AndreiMisiukevich can we just add a comment and maybe a link to some doc, it will help someone figure it out in the future, thanks.
> 
> You mean comment here (Github) or in source code?
> Thx

On the source code.

**Comment by AndreiMisiukevich** [comment]
@rmarinho unfortunately i didn't find any info about size in documentation.
Added comment with pointing to screenshot, which i attached here.

**Pull request merged**